saleor/order/calculations.py

```diff
@@ -0,0 +1,212 @@
+    order.shipping_price = qp(
+        net=tax_data.shipping_price_net_amount,
+        gross=tax_data.shipping_price_gross_amount,
+    )
+    order.shipping_tax_rate = tax_data.shipping_tax_rate
+
+    tax_lines = {line.id: line for line in tax_data.lines}
+    zipped_order_and_tax_lines = ((line, tax_lines[line.id]) for line in lines)
+
+    for (order_line, tax_line) in zipped_order_and_tax_lines:
+        order_line.unit_price = qp(
+            net=tax_line.unit_net_amount, gross=tax_line.unit_gross_amount
+        )
+        order_line.total_price = qp(
+            net=tax_line.total_net_amount, gross=tax_line.total_gross_amount
+        )
+        order_line.tax_rate = tax_line.tax_rate
+
+
+def fetch_order_prices_if_expired(
+    order: Order,
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
+    force_update: bool = False,
+) -> Tuple[Order, Iterable[OrderLine]]:
+    """Fetch order prices with taxes.
+
+    First calculate and apply all order prices with taxes separately,
+    then apply tax data as well if we receive one.
+
+    Prices can be updated only if force_update == True, or if time elapsed from the
+    last price update is greater than settings.ORDER_PRICES_TTL.
+    """
+    if lines is None:
+        lines = list(order.lines.all())
+
+    if not force_update and order.price_expiration_for_unconfirmed < timezone.now():
+        return order, lines
+
+    _apply_tax_data_from_plugins(manager, order, lines)
+
+    tax_data = manager.get_taxes_for_order(order)
+
+    if tax_data:
+        _apply_tax_data(order, lines, tax_data)
+
+    order.price_expiration_for_unconfirmed = timezone.now() + settings.ORDER_PRICES_TTL
+
+    order.save(
+        update_fields=[
+            "total_net_amount",
+            "total_gross_amount",
+            "shipping_price_net_amount",
+            "shipping_price_gross_amount",
+            "shipping_tax_rate",
+            "price_expiration_for_unconfirmed",
+        ]
+    )
+    order.lines.bulk_update(
+        lines,
+        [
+            "unit_price_net_amount",
+            "unit_price_gross_amount",
+            "total_price_net_amount",
+            "total_price_gross_amount",
+            "tax_rate",
+        ],
+    )
+
+    return order, lines
+
+
+def _find_order_line(
+    lines: Iterable[OrderLine],
+    order_line: OrderLine,
+) -> OrderLine:
+    """Return order line from provided lines.
+
+    The return value represents the updated version of order_line parameter.
+    """
+    return next(line for line in lines if line.pk == order_line.pk)
+
+
+def order_line_unit(
+    order: Order,
+    order_line: OrderLine,
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
+    force_update: bool = False,
+) -> TaxedMoney:
+    """Return the unit price of provided line, taxes included.
+
+    Use it instead of calling manager methods directly.
```

Comment:
I don't think that you need to explictly write here `Use it instead...` 
IMO you should just write that we call expected methods on plugins and store the price in the model if prices are expired etc.